storage/db-tool/src/backup_maintenance.rs

```diff
@@ -45,13 +52,14 @@ impl Command {
     pub async fn run(self) -> Result<()> {
         match self {
             Command::Compact(opt) => {
-                let compactor = BackupCompactor::new(
+                let mut compactor = BackupCompactor::new(
```

Comment:
I'd prefer `.run()` to consume itself by taking `mut self` instead of `&mut self`, and you don't need the `mut` here.